zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/CallActivityTransformer.java

```diff
@@ -53,4 +57,19 @@ private void transformProcessId(
         calledElement.isPropagateAllParentVariablesEnabled();
     callActivity.setPropagateAllParentVariablesEnabled(propagateAllParentVariablesEnabled);
   }
+
+  private static void transformLexicographicIndex(
+      final CallActivity element,
+      final List<ExecutableProcess> processes,
+      final ExecutableCallActivity callActivity) {
+    final List<String> allCallActivityIds =
+        processes.stream()
+            .flatMap(p -> p.getFlowElements().stream())
+            .filter(ExecutableCallActivity.class::isInstance)
```

Comment:
❓ Did you consider calculating the index for all elements in the process rather than only call activities?

Reply:
I could. But we actually only need the indexes for call activities, so I thought this will be more performant, especially that we search the list later.

But also, this made it easier to know what to expect in tests.

Reply:
I think as well that it makes more sense, as files could have large numbers of elements. Let's ensure we document the choice somewhere (for example in the commit message).